### ISSUE TITLE
Layout breaks when javascript from other homeservers is blocked

### ISSUE THREAD

**Issue by MorsMortium** [friendica/friendica#14369]
- [x] I have searched open and closed issues for duplicates

### Bug Description

When javascript is blocked from other homeservers, whose posts are displayed, the layout of the posts breaks, and suddenly the banner/top menu row, containing the search bar goes under the posts, when scrolling.

### Steps to Reproduce

1.  Install and enable a js blocking addon: https://noscript.net/
2.  Go on friendica instance, enable the sites own scripts
3.  Log into friendica
4.  See that noscript will also block scripts from other homeservers (mastodon.social, as example)

Actual Result:

When scrolling down, the posts that should disappear under the banner, are going over it, hiding the banner, and only disappear when going out of screen.

Expected Result:

That the posts will still behave correctly, with their homeservers js blocked.

### Screenshots

When logging in:

![image](https://github.com/user-attachments/assets/d8a8298d-8b25-4089-a96e-7e73f57dc365)

Scrolling down:

![image](https://github.com/user-attachments/assets/692735ad-44de-4d72-a590-46b770c51a0a)

Blocked script:

![image](https://github.com/user-attachments/assets/fac2101b-5c92-4f04-98cf-c6cad86abecb)

After enabling it:

![image](https://github.com/user-attachments/assets/42ae6725-db2a-4d8f-ae1b-de022c8ae64b)

### Platform Info

Friendica Version: This is Friendica, version 2024.03 that is running at the web location https://social.trom.tf. The database version is 1557/1557, the post update version is 1544/1550.

Friendica Source: https://github.com/YunoHost-Apps/friendica_ynh

PHP version:

SQL version:

**Comment by bkil** [comment]
Nerdica does not load any third party scripts on my network tab, so it might have something to do with customizations done by Trom.

**Comment by MorsMortium** [comment]
Do you follow all kinds of instances on it? They not always show up either, might be content type dependent, like video

**Comment by bkil** [comment]
Any chance you could give us the URL (pathname) to the blocked resources? It could also be coming from an addon.

> Installed addons/apps: birdavatar, blockem, catavatar, js_upload, langfilter, markdown, newmemberwidget, notifyall, nsfw, opmlexport, pageheader, showmore, startpage, wppost

Also,dos it work correctly if you completely disable JS in your browser?

**Comment by bkil** [comment]
Please check in dev tools whether any errors or warnings have appeared in your console, and also double check further network requests returning non-success or empty response.

**Comment by MorsMortium** [comment]
I checked it now, this was one case it happened.
https://cdn.fosstodon.org/media_attachments/files/113/917/512/934/347/976/original/416220e05e740bdd.mp4 
From this https://fosstodon.org/@Randoneering/113917513142937575
The comment didn't even show with noscript on, seems like the display part kind of crashed.
And no, without js the issue still appears. (with theming also not being available, but that's not related, I think)

**Comment by bkil** [comment]
I think your web browser might embed an mp4 `<video>` player that relies on its own JS engine perhaps. Did you try with another web browser? But also, did you check to see whether there is an option for enabling `<video>` embeds in noscript? ublock and ScriptSafe had options for that I think

**Comment by MorsMortium** [comment]
I use Firefox, as visible from my screenshots. I tested the video here, that is also blocked by noscript https://www.w3schools.com/html/html5_video.asp 
After testing it on chromium, it does the same. There is an option for that, true, but i would need to enable it per provider (per every single home server when I come across a comment from them, on something I follow) And when looking at the video on the w3 website I linked, it didn't break with the video not being available, only displayed the noscript logo, which I could click to re enable that particular video once. This is what I have seen with most websites and what I would expect from friendica too

**Comment by bkil** [comment]
However, now I do see a possible privacy implication. It would be a valid issue to ask whether it would be possible to mirror (cache, proxy) videos similarly to how it is done for media so that the foreign server can't track you or manipulate the served content in a targeted way.

A quick workaround would be to only show such video player after the visitor click a `Play` button (or for nojs, either a details-summary or a checkbox with some CSS or even a subpage of its own with the sole video player).

You can already cope with this client side if you install such a UserCSS yourself (e.g., through Stylus or a UserScript manager).

```
    .video-top-wrapper.lframe:not(:hover) video {
        display: none;
    }
    .video-top-wrapper.lframe {
        min-height: 10em;
    }
    .video-top-wrapper.lframe::before {
        content: "hover to play back media";
        outline: auto;
    }
```

**Comment by bkil** [comment]
I tested it with other JS/video/CSP blocking addons and the layout did not break. I would recommend renaming this issue to
> NoScript breaks layout when a video is blocked on the page

**Comment by MorsMortium** [comment]
That could make the server able to change the content it serves. And it would make the server need to have a huge amount of storage, for such video replies, that I've shown. Having a button to show it would be good, but I think that the rendering of the page breaking is a bug still, and I noticed that the reply also doesn't show

**Comment by MorsMortium** [comment]
> I tested it with other JS/video/CSP blocking addons and the layout did not break. I would recommend renaming this issue to `NoScript breaking layout when a blocked video is on the page`

Can you tell the blockers tested?

**Comment by bkil** [comment]
I've created a new issue to improve privacy protection so you should not need such an extension:
https://github.com/friendica/friendica/issues/14778

I've also linked to another issue that better describes the problem that this one duplicates.

**Comment by MorsMortium** [comment]
That is nice, however I still think this is valid. No other website broke with any removed videos like Friendica

**Comment by bkil** [comment]
Yes, the CSS could be tweaked so that NoScript will break the layout less. That part of issue has already been reported as #12217.

You may consider adding any information missing from there and then closing this issue.
